File: back/server/models/party.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Party struct {
	gorm.Model
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Location    string    `json:"location"`
	Date        time.Time `json:"date"`
	HostID      uint      `json:"host_id"`
	Host        User      `json:"host" gorm:"foreignKey:HostID"`
	Guests      []Guest   `json:"guests" gorm:"foreignKey:PartyID"`
}

File: front/app/models/party.ts
import { Guest } from "./guest";

export interface Party{
  id: number;
  name: string;
  description: string;
  date: string;
  location: string;
  image: string;
  created_at: string;
  updated_at: string;
  hostId: number;
  guests: Guest[];
}